### PR TITLE
Fix Issue #1339 Add pin icon to left hand navigation menu for chat rows that are pinned

### LINKED ISSUE #1339: Add pin icon to left hand navigation menu for chat rows that are pinned
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/master/CONTRIBUTING.md) for onboarding!
___

view the job on Upwork [here](https://www.upwork.com/jobs/~014f4f916839e82c3f).

**Platform - version:**  
all

**Action Performed (reproducible steps):**

1. Log into Expensify Cash
2. Pin a chat using the pin icon at the top right of the chat
image

**Expected Result:**

When a chat is pinned, it should display a small pin icon to the right hand side of the chat row in the left-hand nav.

image

The icon to be used can be found here as an .svg image: 
[pin-circle.svg.zip](https://github.com/Expensify/Expensify.cash/files/5876111/pin-circle.svg.zip)

The icon should use the color `themeColors.icon` and should be displayed at a size of 20x20. 

**Actual Result:**

There is currently no pin icon in the left-hand nav

**Notes/Photos/Videos:** 

Mockup:
105833560-a5396780-5fc9-11eb-9c7c-3ba14c9aaa86

### PR DESCRIPTION
**Pull request by tugbadogan**
cc @jboniface 

### Details
Added pin icon to pinned chats in the left side bar.

### Fixed Issues
Fixes #1339 

### Tests
1. Log into Expensify Cash
2. Pin a chat using the pin icon at the top right of the chat
3. Verify that small pin icon appears on the right hand side of the chat row in the left-hand nav.

### Tested On

- [X] Web
- [X] Mobile Web
- [X] Desktop
- [X] iOS
- [X] Android

#### Web
Screenshot 2021-01-28 at 17 05 44

#### Mobile Web
Screenshot 2021-01-28 at 17 06 30

#### Desktop
Screenshot 2021-01-28 at 17 03 51

#### iOS
Simulator Screen Shot - iPhone 11 - 2021-01-28 at 17 02 23

#### Android
Screenshot_1611853006

### CODE REVIEW

**Comment by tugbadogan** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by shawnborton** [comment]
This looks great! 

For Android, is there any way to create a little more separation between the ellipsis and the icon? Notice the difference between iOS and Android:

Android | iOS
----- | -----
image | image

**Comment by shawnborton** [comment]
Possibly we could add `styles.ml1` (which does `marginLeft: 4px`) to the view that wraps the right icons?

Also, could we see what a chat row looks like when you have entered text into the compose field but you haven't sent it yet? Basically like a "draft" message. Thanks!

**Comment by tugbadogan** [comment]
Hi Shawn,

Thanks for quick review of my pull request.

> Possibly we could add `styles.ml1` (which does `marginLeft: 4px`) to the view that wraps the right icons?

Sure, let me add and see how it looks. 👍
 
> Also, could we see what a chat row looks like when you have entered text into the compose field but you haven't sent it yet? Basically like a "draft" message. Thanks!

Yes, it doesn't look good. 

Screenshot 2021-01-28 at 15 59 41

I noticed that the pencil icon is using `variables.fontSizeSmall` for width and `variables.iconSizeSmall` for height. I think we can make the icon 20x20 (`variables.iconSizeNormal`) like pin icon. It would look like this:

Screenshot 2021-01-28 at 16 01 04

**Comment by shawnborton** [comment]
Agree with all of those thoughts. 

So perhaps we:
- Add `styles.ml1` to the wrapper that contains all of the icons to the right of a chat row
- Inside of this wrapper, we should make the pencil icon use `iconSizeNormal`
- We should also give the pin icon some margin left (maybe `styles.ml2`) to create a gap between the two icons

How does all of that sound?

**Comment by tugbadogan** [comment]
> Agree with all of those thoughts.
> 
> So perhaps we:
> 
> * Add `styles.ml1` to the wrapper that contains all of the icons to the right of a chat row
> * Inside of this wrapper, we should make the pencil icon use `iconSizeNormal`
> * We should also give the pin icon some margin left (maybe `styles.ml2`) to create a gap between the two icons
> 
> How does all of that sound?

I have 2 questions
1. Icon module doesn't use style prop. Do you want me to modify the icon module to accept style prop or wrap the `Icon` with a `View` and add style prop there?
https://github.com/Expensify/Expensify.cash/blob/6424cfdbe990b1e2543c9f75181002cad67e8930/src/components/Icon/index.js#L8-L21
2. Do I need to check whether there is a `Pencil` icon to use `styles.ml2` for the `pin` icon? Otherwise there will be extra margin if there is only `pin` icon.

**Comment by shawnborton** [comment]
1. I think you can just wrap the icon in a view that has margin
2. Great point, so actually, would it just be easier if the wrapper itself had no margin, but every item inside of the wrapper had a margin-left?

**Comment by tugbadogan** [comment]
I added another commit with these changes and updated the screenshots in the first comment.

Screenshot 2021-01-28 at 17 05 44

Screenshot 2021-01-28 at 17 06 30

Screenshot 2021-01-28 at 17 03 51

Simulator Screen Shot - iPhone 11 - 2021-01-28 at 17 02 23

Screenshot_1611853006

**Comment by shawnborton** [comment]
To clarify, is there margin being applied to the left of each icon? I would be curious to see this where each icon (the pin and the pencil) has left margin of 4px, and then another option where each icon has a left margin of 8px.

**Comment by tugbadogan** [comment]
Yes, margins are applied to left of each icon. I adding screenshots with different margins for you to compare. I think 8px margin looks better to me👍

Platform | No Margin | 4px left | 8px left
--- | --- | --- | ---
Web | Screenshot 2021-01-28 at 22 46 53 | Screenshot 2021-01-28 at 17 05 44 | Screenshot 2021-01-28 at 22 49 37
Mobile Web | Screenshot 2021-01-28 at 22 47 51 | Screenshot 2021-01-28 at 17 06 30 | Screenshot 2021-01-28 at 22 48 54
Desktop | Screenshot 2021-01-28 at 22 45 34 | Screenshot 2021-01-28 at 17 03 51 | Screenshot 2021-01-28 at 22 42 30
IOS | Simulator Screen Shot - iPhone 11 - 2021-01-28 at 22 40 34 | Simulator Screen Shot - iPhone 11 - 2021-01-28 at 17 02 23 | Simulator Screen Shot - iPhone 11 - 2021-01-28 at 22 40 56
Android | Screenshot_1611873524 | Screenshot_1611853006 | Screenshot_1611873460

**Comment by shawnborton** [comment]
I agree, 8px looks great so let's go with that option. Thanks so much for exploring!

**Comment by tugbadogan** [comment]
Hi @shawnborton ,

I committed the latest changes and changed margin to 8px.

**Comment by shawnborton** [comment]
Looks good to me, thanks! Looks like there are some conflicts we need to resolve in `ChatLinkRow.js`, then let's give this a final review and get it merged cc @madmax330

**Comment by tugbadogan** [comment]
Hi, I resolved the conflicts and it's ready to merge now.

File: src/pages/home/sidebar/OptionRow.js [hunk 1]
```diff
@@ -138,9 +139,23 @@ const OptionRow = ({
...
+                {option.hasDraftComment && (
+                    <View style={styles.ml2}>
+                        <Icon src={Pencil} width={variables.iconSizeNormal} height={variables.iconSizeNormal} />
+                    </View>
+                )}
+                {option.isPinned && (
+                    <View style={styles.ml2}>
+                        <Icon
+                            src={PinCircle}
+                            width={variables.iconSizeNormal}
```
Comment (shawnborton) [inline src/pages/home/sidebar/OptionRow.js:152]: I think you might be able to omit these declarations as `iconSizeNormal` is the default size of each icon. So could you try without `width`, `height`, and `fill` and see if it looks the same?

**Review by shawnborton — APPROVED** [review]

File: src/pages/home/sidebar/OptionRow.js (hunk 1 above)
Comment (tugbadogan) [reply src/pages/home/sidebar/OptionRow.js]: Yes, removed these properties and it looks the same 👍

**Review by madmax330 — APPROVED** [review]

**Pull request merged**